ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+           real(kind=8), intent(in) :: frac_b(dst_len)
+           real(kind=8), intent(in) :: src(src_len)
+           real(kind=8), intent(in) :: S(n_s)
+           real(kind=8), intent(out) :: dst(dst_len)
+
+           !print*,"Starting bilinear interpolation"
+           !print*,'n_s =',n_s
+           !print*,'src_len =',src_len
+           !print*,'dst_len =',dst_len
+
+           ! Initialize destination field to 0.0
+           do i=1,dst_len
+             dst(i)=0.0
+           enddo
+
+           ! Apply weights
+           do i=1, n_s
+             dst(row(i))=dst(row(i))+S(i)*src(col(i))
+           enddo
+           
+           do i=1, dst_len
+             if (frac_b(i) .ne. 0.0) then
+               dst(i)=dst(i)/frac_b(i)
+             endif
+           enddo
+
+        end subroutine
+    '''
+
+    try:
+        import esmfwghtinterp_f90subroutine
+    except ImportError or ModuleNotFoundError:
+        print('Weighted interpolation fortran subroutine has not been compiled. Compiling...')
+        np.f2py.compile(fsource1, modulename='esmfwghtinterp_f90subroutine', verbose=False)
+        import esmfwghtinterp_f90subroutine
+
+def esmfmanualregrid(src,S,row,col,frac_b,n_s,varmeth,slatd,slond,dlatd,dlond):
+    import esmfwghtinterp_f90subroutine 
+
+    vardims=len(src.shape)
+    if vardims == 4:
+        nlevs=40
+    if vardims == 3:
+        nlevs=1
+
+    dst=np.zeros(shape=(1,nlevs,dlatd,dlond),dtype=float)
+
+    for i in range(0,nlevs): 
+
+        if vardims == 4:
+            src_oned=np.reshape(np.transpose(src[0,i,:,:]),(slatd*slond),'F')
+        if vardims == 3:
+            src_oned=np.reshape(np.transpose(src[0,:,:]),(slatd*slond),'F')
+
+        if 1 == 0:
+            dst_oned= np.zeros(shape=(dlatd*dlond),dtype=float)
+            for m in range(0,n_s):
+                dst_oned[row[m]-1]=dst_oned[row[m]-1]+S[m]*src_oned[col[m]-1]
+
+        if 1 == 1:        
+            dst_oned=esmfwghtinterp_f90subroutine.bilin(src_oned,S,row,col,frac_b,dlatd*dlond)
+
+        dst[0,i,:,:] = np.transpose(np.reshape(dst_oned,(dlond,dlatd),'F'))
+    return dst
+
+def main():
+
+    print('Main program has started')
+
+    #Compile fortran subroutines if they have not already been compiled. 
+    #If already compiled, load fortran-related modules into Python
+    initfortranprog()
+
+    if len(sys.argv) == 6:
+        infiles0=sys.argv[1]
+        outfiles0=sys.argv[2]
+        nemsrc=sys.argv[3]
+        parmfile=sys.argv[4]
+        dstgrds=[sys.argv[5]]
+    else:
+        print('Incorrect number of arguments.')
+        exit()
+
+    if infiles0.endswith('.nc') and outfiles0.endswith('.nc'):
+        print('Input and output files are netcdf.')
+        infiles1=[sys.argv[1]]
+        outfiles1=[sys.argv[2]]
+        dimfiles=1
+    else:
+        with open(infiles0) as f1:
+            infiles1 = f1.read().splitlines()
+        with open(outfiles0) as f2:
+            outfiles1 = f2.read().splitlines()
```

Comment:
Please provide comments or describe what the code/section is doing.
Break code into methods for readability and testing.
Add tests.